client/src/ui/CourseReviews.tsx

```diff
@@ -1,199 +1,30 @@
-import React, { Component, useEffect, useState } from 'react';
-import axios from 'axios';
-import Review from './Review.jsx';
+import React from "react";
+import ReviewCard from "./Review.jsx";
+import { Review } from "common";
 
-import RecentReview from './RecentReview';
-import { Review as ReviewType } from 'common';
-import './css/CourseReviews.css';
-
-type Props = {
-  courseId: string; reviews: readonly ReviewType[]; loading: boolean,
-  onScroll?: any,
-  transformGauges?: any
+type CourseReviewsProps = {
+  reviews: readonly Review[];
+  onReportReview?: (id: string) => void;
 };
-type State = { comparator: 'helpful'; reviews: any };
-
-/*
-  Course Reviews Component.
-
-  Container component that takes in the course's database id and returns:
-  If course Id is valid:
-    a list of all reviews for a course, sorted by date.
-    The component also handles user reporting of reviews.
-
-  If course id = -1:
-    - list of 5 most recent reviews added to the database
-*/
-
-export class CourseReviews extends Component<Props, State> {
-  constructor(props: Props) {
-    super(props);
-    this.state = { comparator: "helpful", reviews: [] };
-  }
-
-  componentDidMount() {
-    axios.post(`/v2/getReviewsByCourseId`, { courseId: this.props.courseId }).then(response => {
-      const reviews = response.data.result
-      if (reviews) {
-        this.setState({ reviews: this.renderReviews(reviews) });
-        this.sort(this.state.comparator);
-      } else {
-        // eslint-disable-next-line no-console
-        console.log(`Unable to find reviews by course by id = ${this.props.courseId}`);
-      }
-    });
-  }
-
-  componentDidUpdate(prevProps: Props) {
-    if (prevProps !== this.props) {
-      axios.post(`/v2/getReviewsByCourseId`, { courseId: this.props.courseId }).then(response => {
-        const reviews = response.data.result
-        if (reviews) {
-          this.setState({ reviews: this.renderReviews(reviews) });
-          this.sort(this.state.comparator);
-        } else {
-          // eslint-disable-next-line no-console
-          console.log(`Unable to find reviews by course by id = ${this.props.courseId}`);
-        }
-      });
-    }
-  }
-
 
-  // Handles selecting different sort bys
-  handleSelect = (event: any) => {
-    let opt = event.target.value;
-    this.setState({ comparator: opt });
-    this.sort(opt);
+export default function CourseReviews({
+  reviews,
```

Comment:
You've probably already seen this, but it seems like ResultsDisplayMobile also uses the CourseReviews component and is currently failing to build since it does not have the logic to pass in a list of reviews. You mentioned in the PR that it might be a good idea to store api calls in a separate file (api.ts) - seems like moving the fetching and sorting logic from ClassView to a file like that could be a possible solution here.

Reply:
good catch - I fixed this now, so hopefully it builds